### PR TITLE
Feat/switch auth flows ts samples

### PR DESCRIPTION
**Pull request by devnotfound**
#### Description of changes:

- This PR adds TypeScript examples for the switching authentication flows category
- This PR also modernises the JavaScript examples for the switching authentication flows category

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [X] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/js/switch-auth.mdx [hunk 1]
```diff
@@ -49,62 +49,164 @@ For more information about working with Lambda Triggers for custom authenticatio
 
 To initiate a custom authentication flow in your app, call `signIn` without a password. A custom challenge needs to be answered using the `sendCustomChallengeAnswer` method:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { CognitoUser } from '@aws-amplify/auth';
```
Comment (cwomack) [inline src/fragments/lib/auth/js/switch-auth.mdx:56]: @devnotfound, we are considering importing CognitoUser directly from the library. Can you remove this line and any references to this type as well?

File: src/fragments/lib/auth/js/switch-auth.mdx [hunk 2]
```diff
@@ -49,62 +49,164 @@ For more information about working with Lambda Triggers for custom authenticatio
...
+
+type SignInParameters = {
+  username: string;
+  password: string;
+};
+
+export async function signIn({ username, password }: SignInParameters) {
+  const challengeResponse = 'the answer for the challenge';
+  try {
+    const user: CognitoUser = await Auth.signIn(username, password);
```
Comment (cwomack) [inline src/fragments/lib/auth/js/switch-auth.mdx:73]: Please remove this reference to CognitoUser

File: src/fragments/lib/auth/js/switch-auth.mdx [hunk 3]
```diff
@@ -49,62 +49,164 @@ For more information about working with Lambda Triggers for custom authenticatio
...
+    if (user?.challengeName === 'CUSTOM_CHALLENGE') {
+      try {
+        // to send the answer of the custom challenge
+        const challengeAnswerResponse: CognitoUser = await Auth.sendCustomChallengeAnswer(
```
Comment (cwomack) [inline src/fragments/lib/auth/js/switch-auth.mdx:77]: And this one

Comment (cwomack) [inline src/fragments/lib/auth/js/switch-auth.mdx:77]: Along with this one, please.

File: src/fragments/lib/auth/js/switch-auth.mdx [hunk 4]
```diff
@@ -49,62 +49,164 @@ For more information about working with Lambda Triggers for custom authenticatio
...
 Here is the sample for creating a CAPTCHA challenge with a Lambda Trigger.
 
 The `Create Auth Challenge Lambda Trigger` creates a CAPTCHA as a challenge to the user. The URL for the CAPTCHA image and  the expected answer is added to the private challenge parameters:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Handler } from 'aws-lambda';
```
Comment (cwomack) [inline src/fragments/lib/auth/js/switch-auth.mdx:137]: @abdallahshaban557, do we want to be importing `Handler` here to offer type support for `'aws-lambda'`? Or do we want to do it as `import type {Handler}`?

Comment (nadetastic) [reply src/fragments/lib/auth/js/switch-auth.mdx:137]: @cwomack I think how it is is fine - https://docs.aws.amazon.com/lambda/latest/dg/typescript-handler.html

Comment (abdallahshaban557) [reply src/fragments/lib/auth/js/switch-auth.mdx:137]: I agree with Dan as well

Comment (cwomack) [reply src/fragments/lib/auth/js/switch-auth.mdx:137]: Thanks @nadetastic for the confirm!

**Review by cwomack — APPROVED** [review]
LGTM

**Review by israx — APPROVED** [review]
LGTM

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by katiegoines — APPROVED** [review]

**Comment by katiegoines** [comment]
@devnotfound Thank you for your contribution to Amplify docs!

**Pull request merged**